pkg/puller/puller.go

```diff
@@ -354,76 +251,68 @@ func (p *Puller) histSyncWorker(ctx context.Context, peer swarm.Address, bin uin
 		s, _, _, err := p.nextPeerInterval(peer, bin)
 		if err != nil {
 			p.metrics.HistWorkerErrCounter.Inc()
-			p.logger.Debug("histSyncWorker nextPeerInterval failed", "error", err)
+			p.logger.Error(err, "histSyncWorker nextPeerInterval failed")
 			return
 		}
+
 		if s > cur {
-			loggerV2.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur)
+			p.logger.Debug("histSyncWorker syncing finished", "bin", bin, "cursor", cur)
 			return
 		}
-		top, ruid, err := p.syncer.SyncInterval(ctx, peer, bin, s, cur)
+		top, _, err := p.syncer.SyncInterval(ctx, peer, bin, s, cur)
 		if err != nil {
```

Comment:
So if there is actual error because of network connection, how would the worker be stopped? This could be multiple workers for each peer. So potentially we have a lot of workers just wasting CPU no?

Reply:
We should be able to test this behaviour. I think this is not trivial to understand from the code, so also we can add some comments about this.

We could test this by calling `ResetPeers` on this mock kademlia. This should return empty for the eachPeerRev calls and trigger the cancellations.